### PR TITLE
Added PDF, PNG, JPEG convenience body types

### PR DESCRIPTION
**Pull request by toastx**
Expanded Body of parser/src/schema/body.rs for body.pdf, body.png, body.jpeg as per #63 

https://github.com/user-attachments/assets/ba53528d-a948-4f02-bdf6-4304c6c19e65

### CODE REVIEW

File: bin/example.toml
Comment (kriogenia) [inline bin/example.toml:1]: remove this file, i gues that it's a leftover for some test that you did

File: bin/tests/inputs/jpeg_post.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[http]
+method = "POST"
+url = "http://127.0.0.1:8000/post"
+
+[headers]
+Content-Disposition = "inline; filename=earth.jpeg"
+Content-Type = "image/jpeg"
+
+[body]
+file = "D:/open-source/rede/bin/tests/assets/earth.jpeg" #modify relative path
```
Comment (kriogenia) [inline bin/tests/inputs/jpeg_post.toml:10]:
```suggestion
file = "./tests/assets/earth.jpeg"
```

File: bin/tests/inputs/pdf_post.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[http]
+method = "POST"
+url = "http://127.0.0.1:8000/post"
+
+[headers]
+Content-Disposition = "inline; filename=dummy-pdf_2.pdf"
+Content-Type = "application/pdf"
+
+[body]
+file = "D:/open-source/rede/bin/tests/assets/dummy-pdf_2.pdf" #modify relative path
```
Comment (kriogenia) [inline bin/tests/inputs/pdf_post.toml:10]:
```suggestion
file = "./tests/assets/dummy-pdf_2.pdf"
```

File: bin/tests/inputs/png_post.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[http]
+method = "POST"
+url = "http://127.0.0.1:8000/post"
+
+[headers]
+Content-Disposition = "inline; filename=moon.png"
+Content-Type = "image/png"
+
+[body]
+file = "D:/open-source/rede/bin/tests/assets/moon.png" #modify relative path
```
Comment (kriogenia) [inline bin/tests/inputs/png_post.toml:10]:
```suggestion
file = "./tests/assets/moon.png"
```

File: parser/src/schema/body.rs [hunk 4]
```diff
@@ -14,6 +14,12 @@ pub(crate) enum Body {
     Json(String),
     #[serde(alias = "xml")]
     Xml(String),
+    #[serde(alias = "pdf")]
+    Pdf(String),
+    #[serde(alias = "png")]
+    Png(String),
+    #[serde(alias = "jpeg")]
```
Comment (kriogenia) [inline parser/src/schema/body.rs:21]:
```suggestion
    #[serde(alias = "jpg", "jpeg")]
```

Let's also support jpg as a way to invoke

File: bin/Cargo.toml [hunk 5]
```diff
@@ -32,13 +32,13 @@ name = "rede"
...
+# rede_parser = "0.2.2"
+# rede_placeholders = "0.1.0"
+# rede_schema = "0.2.0"
 
-# rede_parser = { path = "../parser" }            # local
-# rede_placeholders = { path = "../placeholders"} # local
-# rede_schema = { path = "../schema" }            # local
+rede_parser = { path = "../parser" }             # local
+rede_placeholders = { path = "../placeholders" } # local
+rede_schema = { path = "../schema" }             # local
```
Comment (kriogenia) [inline bin/Cargo.toml:41]: Please, restore it back to its previous state

**Review by kriogenia — CHANGES_REQUESTED** [review]
Hi again, @toastx and thank you so much for taking time to make this awesome contribution. I made a few comments regarding some little details that you probably forgot to fix or restore. The main logic and tests are perfect, so if you fix that I'll merge asap.

If you can't apply those fixes, don't sweat it. Just tell me and I'll do it myself. Thanks again.

**Comment by toastx** [comment]
My bad, i forgot to remove the local dependencies. Also , giving path like that isnt working... it prompts rede to be in the root
Ill change it as per your request tho..

File: bin/example.toml
Comment (toastx) [reply bin/example.toml:1]: that file is present in repo, i did not add it

**Comment by kriogenia** [comment]
> My bad, i forgot to remove the local dependencies. Also , giving path like that isnt working... it prompts rede to be in the root Ill change it as per your request tho..

Don't worry about the request not working right now. It will work once I set-up an integration test when we upgrade the parser version as the tests executed from the `bin` root (you can see in the `body_binary.toml` that is set-up the same way).

**Pull request merged**